utils/src/resource_location.cpp

```diff
@@ -46,19 +39,33 @@ std::string resource_location::to_presentable(bool debug) const
     return utils::path::get_presentable_uri(m_uri, debug);
 }
 
+void resource_location::to_directory()
+{
+    if (!m_uri.empty() && m_uri.back() != '/')
+        m_uri.append("/");
+}
+
 bool resource_location::lexically_out_of_scope() const
 {
     return m_uri == std::string_view("..") || m_uri.starts_with("../") || m_uri.starts_with("..\\");
 }
 
-resource_location resource_location::join(const resource_location& rl, std::string_view relative_path)
+void resource_location::join(std::string_view relative_path)
```

Comment:
I think this should handle the case of appending the `/` in the middle when necessary.

I don't understand why this function cuts off part of the url... it should only be growing, right?

Reply:
As written above (originally based on RFC 3986 but replaced by `relative_reference_resolution()`. Join now behaves in a similar way to `std::filesystem`